server/verify_test.go

```diff
@@ -374,21 +386,30 @@ func TestVerifyBasicAttestationWithTdx(t *testing.T) {
 		TEEDevice: &client.TdxDevice{Device: tdxTestDevice},
 		TEENonce:  nonce64[:],
 	})
-
 	if err != nil {
 		t.Fatalf("failed to attest: %v", err)
 	}
 
 	teeopts := &VerifyTdxOpts{
+		Validation:   TdxDefaultValidateOpts(nonce),
 		Verification: tv.DefaultOptions(),
 	}
 	if _, err := VerifyAttestation(attestation, VerifyOpts{
 		Nonce:      nonce,
 		TrustedAKs: []crypto.PublicKey{ak.PublicKey()},
+		TEEOpts:    teeopts,
 	}); err != nil {
 		t.Errorf("failed to verify: %v", err)
 	}
 
+	if _, err := VerifyAttestation(attestation, VerifyOpts{
+		Nonce:      append(nonce, 0),
+		TrustedAKs: []crypto.PublicKey{ak.PublicKey()},
+		TEEOpts:    teeopts,
```

Comment:
Please also test the code path where we change the *TEE* nonce in addition to the TPM  nonce

Reply:
This code path is tested under `TestVerifyAttestationWithTdx` on line no. 1268 `name: "Wrong TEE Nonce"`